library-js/src/scala/runtime/ScalaRunTime.scala

```diff
@@ -252,22 +252,25 @@ object ScalaRunTime {
       case s => s + "\n"
     }
 
+  // For backwards compatibility with code compiled without -Yexplicit-nulls
+  private inline def mapNull[A, B](a: A, inline f: B): B =
+    if((a: A | Null) == null) null.asInstanceOf[B] else f
+
   // Convert arrays to immutable.ArraySeq for use with Java varargs:
   def genericWrapArray[T](xs: Array[T]): ArraySeq[T] =
```

Comment:
Here we need a big comment with `!!!` explaining what we're doing, and why we chose to do so.

Otherwise, someone will inevitably go through here, think "this is stupid; it can't be null", and "clean up" the code into being wrong.

The same comment applies to all the places where we have done something similar. We need scary `!!!` comments at all the relevant places.